yt/analysis_modules/ppv_cube/ppv_cube.py

```diff
@@ -320,11 +324,17 @@ def __getitem__(self, item):
         return self.data[item]
 
     def _create_intensity(self):
-        def _intensity(field, data):
-            v = self.current_v-data["v_los"].in_cgs().v
-            T = (data["temperature"]).in_cgs().v
-            w = ppv_utils.compute_weight(self.thermal_broad, self.dv_cgs,
-                                         self.particle_mass, v.flatten(), T.flatten())
-            w[np.isnan(w)] = 0.0
-            return data[self.field]*w.reshape(v.shape)
+        if self.thermal_broad:
+            def _intensity(field, data):
+                v = self.current_v-data["v_los"].in_cgs().v
+                T = data["temperature"].in_cgs().v
```

Comment:
Since you're checking specifically for the existence of the temperature field for the gas field type above, it might be worth adding the field type to temperature field query here, and possibly for the `v_los` field as well.